packages/@dcl/inspector/src/components/AssetsCatalog/AssetsCatalog.tsx

```diff
@@ -33,28 +34,30 @@ const AssetsCatalog: React.FC<Props> = ({ catalog }) => {
   )
 
   const filteredCatalog = useMemo(() => {
-    const results = []
+    let results: AssetPack['assets'] = []
+
     if (search) {
-      if (selectedTheme) {
-        for (const asset of selectedTheme.assets) {
-          if (asset.name.toLowerCase().includes(search.toLowerCase())) {
-            results.push(asset)
-          }
-        }
-      } else {
-        for (const theme of catalog) {
-          for (const asset of theme.assets) {
-            if (asset.name.toLowerCase().includes(search.toLowerCase())) {
-              results.push(asset)
-            }
-          }
-        }
+      const assets = selectedTheme ? selectedTheme.assets : catalog.flatMap((theme) => theme.assets)
+      results = assets.filter((asset) => asset.name.toLowerCase().startsWith(search.toLowerCase()))
+
+      if (results.length === 0) {
+        results = assets.filter((asset) => asset.name.toLowerCase().includes(search.toLowerCase()))
```

Comment:
maybe we can do something like:

```typescript
const searchLower = search.toLowerCase()
const { starts, includes } = assets.reduce((results: { starts: AssetPack['assets'], includes: AssetPack['assets'] }, asset) => {
  const name = asset.name.toLowerCase()
  if (name.startsWith(searchLower)) acc.starts.push(asset)
  if (name.includes(searchLower)) acc.includes.push(asset)
  return acc
})

return starts.length ? starts : includes
```

to make it a bit more clear + avoid iterating the same list twice 

(reduce/for...of, etc)